c7n/resources/cloudfront.py

```diff
@@ -181,6 +183,66 @@ def process(self, resources, event=None):
         return results
 
 
+@Distribution.filter_registry.register('check-s3-origin')
+class CheckS3Origin(Filter):
+    """Check for existence of S3 bucket referenced by Cloudfront, and verify ownership.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: cfront-test
+                resource: distribution
+                filters:
+                - type: check-s3-origin
+                  accounts_from:
+                    url: *accounts-list
+                    expr: accounts-list.canonical_id
+    """
+
+    schema = type_schema(
+        'check-s3-origin',
+        accounts_from=ValuesFrom.schema)
+
+    permissions = ('s3:GetBucketAcl',)
+    retry = staticmethod(get_retry(('Throttling',)))
+
+    def process(self, resources, event=None):
+
+        accounts = ValuesFrom(self.data['accounts_from'], self.manager).get_values()
+        results = []
+
+        client = local_session(self.manager.session_factory).client(
+            's3', region_name=self.manager.config.region)
+
+        for r in resources:
+            for x in r['Origins']['Items']:
```

Comment:
split this into another function, the size of the function here is getting pretty long